### PR TITLE
feat: team filters

### LINKED ISSUE #9534: [CAL-1951] No spacing b/w Filter and Event-Types list in smaller view
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1951](https://linear.app/calcom/issue/CAL-1951/no-spacing-bw-filter-and-event-types-list-in-tablet-view)</sub>

Also there is an unnecessary divider(which is only needed in bigger view)

### LINKED ISSUE #9536: [CAL-1953] Use TeamsFilter in Event types and Workflows
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Now that this PR [https://github.com/calcom/cal.com/pull/9417/](https://github.com/calcom/cal.com/pull/9417/) is merged we can reuse TeamsFilter component in event types and workflows

* make changes in the backend route of event types to support filtering

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1953](https://linear.app/calcom/issue/CAL-1953/use-teamsfilter-in-event-types-and-workflows)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/9536
Fixes https://github.com/calcom/cal.com/issues/9534
Fixes https://linear.app/calcom/issue/CAL-1840/org-event-types-filter-backend

 Event types Page
1) When All is selected
2) When Yours is selected
3) When Team Name is selected

https://github.com/calcom/cal.com/assets/53316345/d7fd7e74-c4df-4c98-8ab4-480a37500b86

Responsiveness ( 2nd issue)

https://github.com/calcom/cal.com/assets/53316345/a54d0008-5181-4543-861a-1e98a20cf88c

## Type of change
New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts [hunk 1]
```diff
@@ -137,7 +142,8 @@ export const getByViewerHandler = async ({ ctx }: GetByViewerOptions) => {
   const typesRaw = (
     await prisma.eventType.findMany({
       where: {
-        userId: ctx.user.id,
+        userId: getPrismaWhereUserIdFromFilter(ctx.user.id, input?.filters),
+        team: null,
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts:146]: Not sure about this change. can anyone tell what are we using typesRaw for ?

File: apps/web/pages/event-types/index.tsx [hunk 2]
```diff
@@ -862,6 +858,11 @@ const EventTypesPage = () => {
     );
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
+  const filters = getTeamsFiltersFromQuery(router.query);
+  console.log("filters", filters);
```
Comment (CarinaWolli) [inline apps/web/pages/event-types/index.tsx:862]:
```suggestion
```

File: apps/web/pages/event-types/index.tsx [hunk 3]
```diff
@@ -881,7 +882,7 @@ const EventTypesPage = () => {
           customLoader={<SkeletonLoader />}
           success={({ data }) => (
             <>
-              {data.eventTypeGroups.length > 1 ? (
+              {data.eventTypeGroups.length > 1 && (
```
Comment (CarinaWolli) [inline apps/web/pages/event-types/index.tsx:885]: I would prefer showing the `EventTypeListHeading` also if we have a filtered view with only one checked item
Screenshot 2023-06-19 at 09 28 45

**Review by CarinaWolli — DISMISSED** [review]
The filter has some issues on mobile, but I think we don't even need the filter on mobile as we anyways show the filtered tabs

File: apps/web/pages/event-types/index.tsx (hunk 3 above)
Comment (CarinaWolli) [reply apps/web/pages/event-types/index.tsx]: @Udit-takkar Maybe wasn't fully clear here. I meant to only show the heading if the user also has teams (but it's filtered to only one). If a user does not have teams we should show the heading as per design. If that would mean we need to fetch some extra data, we can also leave it without the heading (as you had it before)

Comment (Udit-takkar) [reply apps/web/pages/event-types/index.tsx]: @CarinaWolli if a user does not have team then we do not show heading for user event types (in the design)

**Review by PeerRich — DISMISSED** [review]

File: apps/web/pages/event-types/index.tsx (hunk 3 above)
Comment (CarinaWolli) [reply apps/web/pages/event-types/index.tsx]: But the heading is currently always shown, also without teams:
Screenshot 2023-06-19 at 14 50 47

Comment (Udit-takkar) [reply apps/web/pages/event-types/index.tsx]: @CarinaWolli pushed some changes fixing this

**Review by CarinaWolli — APPROVED** [review]
LGTM 👍

**Review by emrysal — APPROVED** [review]
looks good, the typesRaw is something backwards compatible, not sure if there's still reason but assume it does.

**Pull request merged**